`eth_getCode` and `eth_getStorageAt` unable to deal with non-EVM actors

These methods currently return errors if they're called on non-EVM actors (e.g. accounts or else). Instead, they should return empty success responses. See Lotus:

https://github.com/filecoin-project/lotus/blob/b0bc4a96320942c0673a3cab08314074edf4b4f9/node/impl/full/eth.go#L450

Here are steps to reproduce:


Launch a single Fendermint node:

```bash
cd fendermint/fendermint/testing/smoke-test
cargo make setup
```

In another terminal, clone the Fluence repository:

```bash
cd fluence
git clone https://github.com/fluencelabs/deal -b test-ipc
cd deal
git submodule update --init --recursive
```

Create a `.env` file with the private key from one of the Ethereum accounts which were added to `genesis.json`:

```bash
cat << EOF > .env
PRIVATE_KEY=0x$(cat ../../ipc-monorepo/fendermint/fendermint/testing/smoke-test/test-data/keys/eric-eth.sk)
EOF
```

Try to deploy:

```bash
make deploy-local-ipc
```

```
$ make deploy-local-ipc
...

2024-01-04T13:05:36.741114Z ERROR sharedbackend: Failed to send/recv `basic` err=GetAccount(0xa915ea7780d80adb0c4595952a2dbe115e4fe504, 
(code: 22, message: message failed with backtrace:
00: f0105 (method 3) -- invalid method: 3 (22)
, data: None)) address=0xa915ea7780d80adb0c4595952a2dbe115e4fe504
Error: 
Failed to get account for 0xa915ea7780d80adb0c4595952a2dbe115e4fe504: 0xa915ea7780d80adb0c4595952a2dbe115e4fe504
make: *** [Makefile:45: deploy-local-ipc] Error 1
```

--- a/fendermint/fendermint/eth/api/examples/ethers.rs
+++ b/fendermint/fendermint/eth/api/examples/ethers.rs
@@ -601,18 +601,23 @@ where
     // We could calculate the storage location of the balance of the owner of the contract,
     // but let's just see what it returns with at slot 0. See an example at
     // https://ethereum.org/en/developers/docs/apis/json-rpc/#eth_getstorageat
+    let storage_location = {
+        let mut bz = [0u8; 32];
+        U256::zero().to_big_endian(&mut bz);
+        H256::from_slice(&bz)
+    };
+
     request(
         "eth_getStorageAt",
-        mw.get_storage_at(
-            contract.address(),
-            {
-                let mut bz = [0u8; 32];
-                U256::zero().to_big_endian(&mut bz);
-                H256::from_slice(&bz)
-            },
-            None,
-        )
-        .await,
+        mw.get_storage_at(contract.address(), storage_location, None)
+            .await,
+        |_| true,
+    )?;
+
+    request(
+        "eth_getStorageAt /w account",
+        mw.get_storage_at(from.eth_addr, storage_location, None)
+            .await,
         |_| true,
     )?;
 

--- a/fendermint/fendermint/eth/api/examples/ethers.rs
+++ b/fendermint/fendermint/eth/api/examples/ethers.rs
@@ -617,9 +617,15 @@ where
     )?;
 
     request(
-        "eth_code",
+        "eth_getCode",
         mw.get_code(contract.address(), None).await,
         |bz| *bz == deployed_bytecode,
+    )?;
+
+    request(
+        "eth_getCode /w account",
+        mw.get_code(from.eth_addr, None).await,
+        |bz| bz.is_empty(),
     )?;
 
     request("eth_syncing", mw.syncing().await, |s| {

--- a/fendermint/fendermint/eth/api/src/state.rs
+++ b/fendermint/fendermint/eth/api/src/state.rs
@@ -371,7 +371,14 @@ where
             .context("failed to call contract")?;
 
         if result.value.code.is_err() {
-            return error(ExitCode::new(result.value.code.value()), result.value.info);
+            return match ExitCode::new(result.value.code.value()) {
+                ExitCode::USR_UNHANDLED_MESSAGE => {
+                    // If the account is an ETHACCOUNT then it doesn't handle certain methods like `GetCode`.
+                    // Let's make it work the same way as a PLACEHOLDER and return nothing.
+                    Ok(None)
+                }
+                other => error(other, result.value.info),
+            };
         }
 
         tracing::debug!(addr = ?address, method_num, data = hex::encode(&result.value.data), "evm actor response");

